### PR TITLE
Implement analytically linearized Poisson Boltzmann model

### PR DESCRIPTION
**Pull request by awvwgk**
Adds the ALPB solvation model. Only minor modifications in the GB model are required to implement the new model, therefore GB and ALPB will be implemented by the same class.

### CODE REVIEW

File: prog/dftb+/lib_solvation/born.F90 [hunk 1]
```diff
@@ -1051,4 +1086,135 @@ subroutine getBornEGCluster(self, coords, energies, gradients, sigma)
...
+    real(dp) :: r2, rad2, rad3, totRad3, vec(3), center(3), inertia(3, 3), aDet
+    real(dp) :: aDeriv(3, 3), qtotal
+    real(dp), parameter :: tof = 2.0_dp/5.0_dp, unity(3, 3) = reshape(&
+      & [1.0_dp, 0.0_dp, 0.0_dp, 0.0_dp, 1.0_dp, 0.0_dp, 0.0_dp, 0.0_dp, 1.0_dp], &
+      & [3, 3])
+
+    qtotal = 0.0_dp
+    totRad3 = 0.0_dp
+    center(:) = 0.0_dp
+    do iAt = 1, nAtom
```
[the remark this answers is not in the document]
Comment (awvwgk) [reply prog/dftb+/lib_solvation/born.F90:1200]: Can't be reused, as we calculate the total charge in the gradient but not in the potential terms and I want to save one loop over all atoms here.

File: prog/dftb+/lib_solvation/born.F90 [hunk 2]
```diff
@@ -1051,4 +1086,135 @@ subroutine getBornEGCluster(self, coords, energies, gradients, sigma)
...
+      iSp = species(iAt)
+      rad2 = rad(iSp) * rad(iSp)
+      rad3 = rad2 * rad(iSp)
+      totRad3 = totRad3 + rad3
+      center(:) = center + coord(:, iAt) * rad3
+      qtotal = qtotal + qvec(iAt)
+    end do
+    center = center / totRad3
+
+    inertia(:, :) = 0.0_dp
```
[the remark this answers is not in the document]
Comment (awvwgk) [reply prog/dftb+/lib_solvation/born.F90]: Factored out as `getInertia`.

File: prog/dftb+/lib_solvation/born.F90 [hunk 3]
```diff
@@ -1051,4 +1086,135 @@ subroutine getBornEGCluster(self, coords, energies, gradients, sigma)
...
+    !> Atomic gross charges
+    real(dp), intent(in) :: qvec(:)
+
+    !> Molecular gradient
+    real(dp), intent(inout) :: gradient(:, :)
+
+    integer :: iAt, iSp
+    real(dp) :: r2, rad2, rad3, totRad3, vec(3), center(3), inertia(3, 3), aDet
+    real(dp) :: aDeriv(3, 3), qtotal
+    real(dp), parameter :: tof = 2.0_dp/5.0_dp, unity(3, 3) = reshape(&
```
[the remark this answers is not in the document]
Comment (awvwgk) [reply prog/dftb+/lib_solvation/born.F90]: Moved to `getInertia` and therefore only needed once now.

**Comment by bhourahine** [comment]
Good to go, merging now.

**Pull request merged**